crates/store/src/db/models/queries/accounts.rs

```diff
@@ -915,3 +924,169 @@ pub(crate) struct AccountStorageMapRowInsert {
     pub(crate) value: Vec<u8>,
     pub(crate) is_latest_update: bool,
 }
+
+// CLEANUP FUNCTIONS
+// ================================================================================================
+
+/// Clean up old vault asset entries for a specific account, keeping only the latest entry
+/// (is_latest_update=true) and up to MAX_HISTORICAL_ENTRIES_PER_ACCOUNT older entries per
+/// (account_id, vault_key) combination.
+///
+/// # Parameters
+/// * `conn`: Database connection
+/// * `account_id`: The account to clean up
+///
+/// # Returns
+/// The number of rows deleted
+///
+/// # Notes
+/// This function ensures we don't accumulate unbounded historical data while preserving:
+/// - The latest state for each vault key (is_latest_update=true)
+/// - Up to MAX_HISTORICAL_ENTRIES_PER_ACCOUNT historical entries per vault key
+pub(crate) fn cleanup_old_account_vault_assets(
+    conn: &mut SqliteConnection,
+    account_id: AccountId,
+) -> Result<usize, DatabaseError> {
+    // Strategy: For each vault_key, keep the latest entry (is_latest_update=true) and the
+    // most recent MAX_HISTORICAL_ENTRIES_PER_ACCOUNT historical entries, deleting the rest.
+    //
+    // Unfortunately, Diesel doesn't support window functions (ROW_NUMBER OVER) in its type-safe
+    // API, so we must use raw SQL for this complex deletion pattern. This is necessary because:
+    // 1. We need to partition by vault_key and order by block_num
+    // 2. We need to delete based on row ranking within each partition
+    // 3. The table is created WITHOUT ROWID, so we use the primary key (account_id, block_num,
+    //    vault_key)
```

Comment:
I'm not sure we need to keep `MAX_HISTORICAL_ENTRIES_PER_ACCOUNT` - we just need to make sure we have the data for the last $n$ blocks. So, query could be as simple as:
```sql
DELETE FROM account_vault_assets
WHERE
    block_num < chain_tip - n
    AND is_latest_update = false
```
To make this query efficient, we may need to add indexes on `block_num` and `is_latest_update` fields.

This methodology should also apply to the `account_storage_map_values` table (and in the future to the `accounts` table).